code/src/containers/forms/product-form/ProductForm.types.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Control,
+  FieldErrors,
+  UseFormRegister,
+  UseFormRegisterReturn
+} from "react-hook-form";
+
+import { AppInputProps } from "@/components/app-input/AppInput.types";
+
+import { Locale } from "@/context/i18n/I18nProvider";
+import { FullManagerProduct } from "@/types/product.types";
+
+export type ProductFormValues = {
+  status: boolean;
+  image: string;
+  quantity: number | undefined;
+  price: number | undefined;
+  category: number;
+  productTranslations: ProductTranslation[];
+};
+
+export type UpdateProductFormValues = {
+  status: boolean;
+  image: string;
+  quantity: number | undefined;
+  price: number | undefined;
+  tagIds: number[];
+  productTranslations: ProductTranslation[];
+};
+
+export type ProductTranslation = {
+  name: string;
+  description: string;
+  languageCode: Locale;
+};
+
+export type ImagePreviewProps = {
+  imageInputProps: UseFormRegisterReturn<"image"> & AppInputProps;
+  defaultValue?: string;
+};
+
+export type UpdateProductFormProps = {
+  product: FullManagerProduct;
+};
+
+export type ProductFormRegisterFunction = UseFormRegister<ProductFormValues>;
+export type ProductFormFieldErrors = FieldErrors<ProductFormValues>;
+export type ProductFormControl = Control<ProductFormValues>;
+
+export type ProductFormAdditionalInfoSectionProps = {
+  register: ProductFormRegisterFunction;
+  errors: ProductFormFieldErrors;
+  control: ProductFormControl;
+};
+
+export type ProductFormImageSectionProps = {
+  register: ProductFormRegisterFunction;
+  errors: ProductFormFieldErrors;
+  control: ProductFormControl;
+};
+
+export type ProductFormMainInfoSectionProps = {
+  register: ProductFormRegisterFunction;
+  errors: ProductFormFieldErrors;
+};
```

Comment:
Same here

Reply:
Fixed, I created base type and extended from it